packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -259,14 +259,15 @@ class AndroidDevice extends Device {
     return true;
   }
 
-  Future<int> _forwardPort(String service, int devicePort, int port) async {
+  Future<Uri> _forwardUriToNewPort(String service, Uri deviceUri, int port) async {
```

Comment:
I suggest keeping the original name signature of this function.  The name is confusing because we cannot forward a "uri" to a "port". Just change the callers to pass in uri.port.